server/utils/socketServer.js

```diff
@@ -0,0 +1,47 @@
+const jwt = require("jsonwebtoken");
+const User = require('../models/User');
+const Profile = require('../models/Profile');
+const notificationContent = require('../templates/notificationTemplates');
+const onlineUsers = new Map();
+
+const getId = (cookie) => {
+  const tempArrayOne = cookie.split('=');
+  const tempArrayTwo = tempArrayOne[1].split(';');
+  const token = tempArrayTwo[0];
+  const id = jwt.verify(token, process.env.JWT_SECRET).id;
+  return id;
+};
+
+const sock = (io) => {
+  io.on('connection', async (socket) => {
+    socket.on('authenticate', async () => {
+      const cookie = socket.handshake.headers.cookie;
+      const id = getId(cookie);
+      const user = await User.findById(id);
+      if (user) {
+        onlineUsers.set(id, { ...user._doc, socketId: socket.id });
+        return onlineUsers;
+      }
+    });
+    console.log("connected");
```

Comment:
I'm fine with the logging but could we make this more clear like "Socket connection established for {user id}" 🤔

Reply:
I had that and was asked to remove them 
Look at previous comments